lib/XML/LibXML/CStructs.pm

```diff
@@ -132,6 +159,34 @@ my class xmlNode is export(:types) {
     has uint16 $.line;
     #~ unsigned short	extra	: extra data for XPath/XSLT
     has uint16 $.extra;
+
+
+    # cw: Looks like having methods in these classes will be unavoidable unless we 
+    #     decide to make the attributes rw... which might be workable as long as 
+    #     we can protect them in descendent classes
+    method setNext(xmlNodePtr $n) {
```

Comment:
either for all or for none, no? I'd like to see an interface where you dont have to remember what attrs are writeable and what attrs have setters.